### DISCUSSION TITLE
Raspi  0 W 2 wifi extender based on 2 wifi usb dongles doesn't send raspi-gui on the air

### DISCUSSION THREAD

**Discussion by Fabri54** [RaspAP/raspap-webgui#1550] [Q&A]
I've built a WiFi extender based on a raspi Zero W 2, its on-board  wifi chip and a WiFi USB dongle  by BrosTrend and, of course, RaspAP. It works fine, without a glitch except for the speed that is a bit low but, for my purpose, it's enough.
However, I want to replace the on-board WiFi  with another WiFi USB dongle, similar to the previous one, for a matter of coverage. My Pi Zero W has an USB-Ethernet hat that gives me 3 USB ports and a very compact form factor.
Now, if I disable the on-board WiFi chip (blacklist brcmfmac) add SSID, passoword and country, using raspi-config, detach the ethernet cable and reboot, I end with the following ifconfig:

```
eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether 00:e0:4c:36:00:9a  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 6  bytes 368 (368.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 6  bytes 368 (368.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

wlan0: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 10.3.141.1  netmask 255.255.255.0  broadcast 10.3.141.255
        inet6 fe80::fa79:20e3:d3a5:1f74  prefixlen 64  scopeid 0x20<link>
        ether 34:7d:e4:4f:b0:67  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 114 overruns 0  carrier 0  collisions 0

wlan1: flags=4163<UP,BROADCAST,RUNNING,MULTICAST>  mtu 1500
        inet 192.168.1.44  netmask 255.255.255.0  broadcast 192.168.1.255
        inet6 fe80::d15b:4de:7f5c:cbb3  prefixlen 64  scopeid 0x20<link>
        ether 34:7d:e4:4f:bb:51  txqueuelen 1000  (Ethernet)
        RX packets 3298  bytes 478216 (467.0 KiB)
        RX errors 0  dropped 3  overruns 0  frame 0
        TX packets 886  bytes 117866 (115.1 KiB)
        TX errors 0  dropped 21 overruns 0  carrier 0  collisions 0
```

I can then log in using SSH via the connection to the main WLAN (wlan1)  but there is no trace of SSID raspi-webgui on the air, anywhere. I'm obviously doing something wrong but I don't know what. Where should I look (configuration files or else) to  try to understand and solve this issue?
Many thanks for any help

**Comment by billz** [comment]
As always, the [debug log generator](https://docs.raspap.com/ap-basics/#debug-log) will give you the best starting point for troubleshooting.

**Reply by Fabri54** [reply]
OK, I've done it. Is there any tutorial or guideline on how to read and interpret the debug.log? Please note that The problem I've is to understand why the two WiFi dongles are seen (lsusb) , both are given the proper IP address (192.168.1.xy the one connected to the home AP, and 10.3.141.1 the one that should  transmit the raspi-webgui) and, apparently, are also transmitting with a certain power level but, as I wrote, there is no sign of the SSID raspi-webgui on the air. Any hint about where is should look?
Many thanks in advance for any help

**Reply by Fabri54** [reply]
P.S. I've also copied the debug.log file on pastebin.com but, not having any familiarity with the tool, I don't know ho to link it into this post

**Comment by Fabri54** [comment]
Eventually, for lack of knowledge and understanding, I gave up the analysis of the debug.log and, after some empirical trials , I've come to a solution and got a working WiFi Extender based on a Raspberry Pi Zero W 2 and two WiFi  USB dongles (BrosTrend AC805) , being the on-board WiFi chip disabled (blacklist brcmfmac). 
I do not know why the procedure works, what the is the rationale and if it's repetitive on other different devices.
I post it here because it might help somebody but, please, do not take it for solid gold and, above all, consider that I cannot provide 
any support should something  go wrong with your installation as I don't have the faintest idea of the reasons why it works.

* I started from a  working WiFi Extender based on on-board chip (10.3.141.1/24)  and one WiFi USB dongle (192.168.1.wy).

* I've added a second WiFi USB dongle  that was recognized and utilized as a second connection to the local WLAN. So, I had the on-board WiFi chip that got a 10.3.141.1/24 IP address, a first WiFi USB dongle that got a 192.168.1.wy IP address and the second WiFi USB dongle that got another 192.168.1.yz ID address.

* Subsequently I blacklisted the on-board chip and rebooted the Raspi.

* Doing so I got something connected to the local WLAN via wlan0 and a wlan1 with an assigned IP 0f 10.3.141.1/24. I could log in via SSH and manage it but no way of finding  the SSID raspi-gui on the air.

* I tried to unplug and plug again one of the WiFi dongles but nothing changed (it was connected to theW LAN, 10.3.141,1/24 IP assigned to the other WiFi, no raspi-gui on the air)

*  Just out of desperation I removed one of the WiFi dongles , I shutdown, powered off the "thing" and restarted.

*  When it was running stable, I plugged in the second WiFi dongle and the SSID raspi-gui materialized in the air. I could connect via my iPad to raspi-webgui. Changed passwords, changed the metric, got internet connection and that was it.

* Since yesterday it has stood several shutdowns, movement to another part of the flat and restart. The procedure of hot-plugging the second WiFi dongle seems to work stable and the speed of the connection is enough to let me watch a Netflix movie on my iPad connected to raspi-webgui.

So, it works and I hope, one day or another to be able to understand why ....
